src/java_bytecode/library/Makefile

```diff
@@ -1,26 +1,51 @@
-all: org.cprover.jar
+.NOTPARALLEL:
+#javac compiles multiple classes for each source as it will compile dependent sources.
+#Thus we do not allow the make to run concurrently.
+
+SRC = converter.cpp
+
+include ../../config.inc
+include ../../common
 
 SOURCE_DIR := src
 BINARY_DIR := classes
 
 FIND := find
 
 JAVAC := javac
-JFLAGS := -sourcepath $(SOURCE_DIR) -d $(BINARY_DIR)
+JFLAGS := -sourcepath $(SOURCE_DIR) -d $(BINARY_DIR) -XDignore.symbol.file
 
 CLASSPATH := SOURCE_DIR
 
-all_javas := $(shell $(FIND) $(SOURCE_DIR) -name '*.java')
+ALL_JAVAS := $(wildcard $(SOURCE_DIR)/*/*.java $(SOURCE_DIR)/*/*/*.java $(SOURCE_DIR)/*/*/*/*.java)
+ALL_CLASSES := $(patsubst $(SOURCE_DIR)/%.java,$(BINARY_DIR)/%.class,$(ALL_JAVAS))
 
 $(BINARY_DIR):
 	mkdir -p $(BINARY_DIR)
 
-.PHONY: compile
-compile: $(BINARY_DIR)
-	$(JAVAC) $(JFLAGS) $(all_javas)
+.SUFFIXES: .java .class
+
+$(BINARY_DIR)/%.class : $(SOURCE_DIR)/%.java $(BINARY_DIR)
+	$(JAVAC) $(JFLAGS) $(patsubst $(BINARY_DIR)/%.class,$(SOURCE_DIR)/%.java,$@)
 
 JAR := jar
 JARFLAGS := -cf
 
-org.cprover.jar: compile
-	$(JAR) $(JARFLAGS) $@ -C $(BINARY_DIR) org
+core-models.jar: $(ALL_CLASSES)
+	$(JAR) $(JARFLAGS) $@ -C $(BINARY_DIR) .
+
+CLEANFILES = core-models.jar converter$(EXEEXT) java_core_models.inc
+
+clean : clean_
+
+clean_ :
+	$(RM) -Rf $(BINARY_DIR)
+
+all: java_core_models.inc
```

Comment:
Please make the `all` goal the first one in this file to make it the default.

Reply:
done